### PR TITLE
feat(ray): Remove initContainers

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Tracking issue
Resolves: flyteorg/flyte#5165

## Why are the changes needed?

[The init container in flyte](https://github.com/flyteorg/flyte/blob/9cf1f6fe67f3804178bbf39765f15aca39ce7d27/flyteplugins/go/tasks/plugins/k8s/ray/ray.go#L383) is no longer needed because [the latest kuberay already has init container](https://github.com/ray-project/kuberay/blob/ec40186fed77eba2a05612248fa3a2ef889ecd2b/ray-operator/controllers/ray/common/pod.go#L178).

## What changes were proposed in this pull request?

Remove the init container in flyte.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

**Comment by MortalHappiness** [comment]
@ByronHsu Can you help me review this PR? Thanks.

**Comment by ByronHsu** [comment]
Looks good. Have you run e2e test? Can you attach a successful run screenshot?

**Comment by MortalHappiness** [comment]
> Looks good. Have you run e2e test? Can you attach a successful run screenshot?

image

File: go.mod [hunk 1]
```diff
@@ -5,8 +5,9 @@ go 1.21
 require (
 	github.com/flyteorg/flyte/datacatalog v0.0.0-00010101000000-000000000000
 	github.com/flyteorg/flyte/flyteadmin v0.0.0-00010101000000-000000000000
-	github.com/flyteorg/flyte/flytepropeller v0.0.0-00010101000000-000000000000
-	github.com/flyteorg/flyte/flytestdlib v0.0.0-00010101000000-000000000000
+	github.com/flyteorg/flyte/flytepropeller v1.9.12
+	github.com/flyteorg/flyte/flytestdlib v1.9.12
+	github.com/flyteorg/flytectl v0.8.15
```
Comment (pingsutw) [inline go.mod:10]: We shouldn't update this.

**Review by pingsutw — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (eapolinario) [reply go.mod:10]: Just to clarify, leaving the [pseudo-versions](https://go.dev/ref/mod#pseudo-versions) help in show that those are replaced at the [end of this file](https://github.com/flyteorg/flyte/blob/master/go.mod#L234-L240).

Comment (MortalHappiness) [reply go.mod:10]: @eapolinario But I run `make go-tidy` on master branch and these lines of pseudo-versions changed. I do not change them manually. Do you know what happened? Is this a special golang behaviour?

Comment (MortalHappiness) [reply go.mod:10]: image

Comment (MortalHappiness) [reply go.mod:10]: By the way, @pingsutw said that he also ran `make go-tidy` on the latest commit on the master branch but those pseudo-versions do not change.

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**